### PR TITLE
[Go] Modules sharing packages example

### PR DESCRIPTION
**Pull request by DazWilkin**
@ImJasonH  -- added an example.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/DazWilkin/cloud-builders
```
Comment (ImJasonH) [inline go.mod]: Is this file necessary? It should be enough to have the go.mod in `go/examples/multi_step/` right?

Comment (DazWilkin) [reply go.mod]: Hmmm... user error.

Comment (DazWilkin) [reply go.mod]: Apologies. Fixed: https://github.com/GoogleCloudPlatform/cloud-builders/pull/518/commits/ecb516fc9775eed1d5c9926c4a71054765569c3e
